### PR TITLE
feat: add minute second inputs for offset

### PR DESCRIPTION
**Pull request by jackblatch**
This changes the offset input to two inputs allowing for minutes and seconds to be entered, rather than just seconds.
#41

### CODE REVIEW

**Comment by markflorkowski** [comment]
Would you mind including a screenshot since this is a visual change?

File: src/app/(core)/v/[slug]/player.tsx [hunk 1]
```diff
@@ -132,14 +136,30 @@ export const VodPlayer = (props: { id: string; vod: VOD }) => {
...
             onChange={(e) => {
-              setOffset(parseInt(e.currentTarget.value, 10));
+              setOffset(prev => {
+                const value: number = parseInt(e.target?.value, 10);
+                return {...prev, minutes: !value && value !== 0 ? "" : value, totalSeconds: (typeof value === 'number' ? value * 60 : 0) + (typeof prev.seconds === 'number' ? prev.seconds : 0)}
+              });
             }}
             suffixEl={<span className="text-gray-300">{`seconds`}</span>}
           />
+                  <TextInput
```
Comment (markflorkowski) [inline src/app/(core)/v/[slug]/player.tsx]: This should come before the seconds field if we do it this way. 

I think I would prefer a single input that takes either seconds *or*  hh:mm:ss though, and it converts to seconds. See discussion in #36

**Comment by jackblatch** [comment]
@markflorkowski I've added the feature into a single input now. Let me know what you think. Do you have any preference on what the input suffix should say with this update (as it's not always shown as seconds anymore)?

offset input

**Comment by markflorkowski** [comment]
> @markflorkowski I've added the feature into a single input now. Let me know what you think. Do you have any preference on what the input suffix should say with this update (as it's not always shown as seconds anymore)?
> 
> 
> 
> offset input
> 
> 

I think we can either 

A) remove the suffix 

or 

B) hide it as soon as the user enters a `:` in the field

Personally leaning towards (A), @t3dotgg do you have an opinion?

File: src/app/(core)/v/[slug]/player.tsx [hunk 2]
```diff
@@ -86,18 +86,33 @@ export const VodPlayer = (props: { id: string; vod: VOD }) => {
...
 
-    const startTime = Math.max(marker.position_seconds - offset, 0);
+    const startTime = Math.max(marker.position_seconds - offset.totalSeconds, 0);
 
     return `${startTime},${endTime},${marker.description.replace(",", "")}`;
   });
 
+  function parseOffsetValue(value: string): number | undefined {
+    // if there are no colons, assume its seconds
+    if(/^\d+$/.test(value)) return parseInt(value, 0);
```
Comment (flbn) [inline src/app/(core)/v/[slug]/player.tsx]: `parseInt` should prob have a radix of 10. as per [browser standards](https://262.ecma-international.org/13.0/#sec-parseint-string-radix), adding a 0 just let's the browser know you want the default radix (which, according to this is supposed to be 10 if the browser is ECMA compliant but might as well just put the expected value)

```suggestion
    if(/^\d+$/.test(value)) return parseInt(value, 10);
```

**Comment by markflorkowski** [comment]
Talked to Theo, he seems in agreement that removing the suffix is sufficient.

**Comment by jackblatch** [comment]
> 

Sure, that's done now.

File: src/app/(core)/v/[slug]/player.tsx [hunk 3]
```diff
@@ -86,18 +86,33 @@ export const VodPlayer = (props: { id: string; vod: VOD }) => {
     ...props.vod.markers,
   ];
 
-  const [offset, setOffset] = useState(0);
-
+  const [offset, setOffset] = useState<{presentational: string, totalSeconds: number}>({
```
Comment (markflorkowski) [inline src/app/(core)/v/[slug]/player.tsx:89]: this breaks output csv file names, need to change L135 to reflect this

Comment (jackblatch) [reply src/app/(core)/v/[slug]/player.tsx:89]: good catch, should be fixed now.

**Review by markflorkowski — APPROVED** [review]

**Pull request merged**